src/subs/WCSFoFRoutine.cpp

```diff
@@ -346,8 +346,17 @@ void FoFClass::sortMatches(int fieldNumber, int minMatches, bool allowSelfMatche
         if (pcat.empty()) continue;
 
         long matches = 0;
+
+        // Sort the matches by their ra position to ensure deterministic behavior downstream.
+        const auto matchSort = [](fof::Match<Point, 2> *a, fof::Match<Point, 2> *b) {
+            return a->getAverage()[0] < b->getAverage()[0];
```

Comment:
If you only care about the average RA, you might want to add a TODO to support getting the average for one dimension, again assuming that's a worthwhile optimization.

Reply:
Same as above, we don't need to worry about optimization since this will be replaced soon.